src/ucs/sys/sys.h

```diff
@@ -362,6 +362,45 @@ void *ucs_sys_realloc(void *old_ptr, size_t old_length, size_t new_length);
  */
 void ucs_sys_free(void *ptr, size_t length);
 
+
+/**
+ * Detect memory nodes associated with the CPUs and GPUs on the system
+ * and return a list of such nodes
+ *
+ * @param [out] mm_units     Array of memory unit abstractions
+ * @param [out] num_mm_units Number of memory units detected in the system
+ * @return UCS_OK or error in case of failure.
+ */
+ucs_status_t ucs_sys_get_mm_units(ucs_mm_unit_t **mm_units, int *num_mm_units);
```

Comment:
i don't agree with this API.
need only:
1. API to get device handle (ucs_sys_device_t) by name (could be GPU or HCA name for example ) - to be used by UCTs
2. API to get latency and BW between 2 devices by their handles
3. optional helper API to get device handle of local CPU NUMA node (for CPU memory transports)

UCTs will have API to accept pointer and return ucs_sys_device_t which this pointer belongs to (using API (1) )
UCP will use API (2) to tune rma_bw_lanes